### PR TITLE
option to return :first_throttled when the throttle is hit the first …

### PR DESCRIPTION
**Pull request by grosser**
…time

can be used to send notifications only on first throttle breach

@ggrossman @melvinram 

### Risks
 - None

### CODE REVIEW

File: lib/prop/limiter.rb [hunk 1]
```diff
@@ -85,7 +91,7 @@ def throttle(handle, key = nil, options = {}, &block)
       # Returns the value of the block if given a such, otherwise the current count of the throttle
       def throttle!(handle, key = nil, options = {}, &block)
         options, cache_key = prepare(handle, key, options)
-        throttled, counter = _throttle(handle, key, options, &block)
+        throttled, counter = _throttle(handle, key, cache_key, options, &block)
 
         if throttled
           raise Prop::RateLimited.new(options.merge(cache_key: cache_key, handle: handle))
```
Comment (ggrossman) [inline lib/prop/limiter.rb]: Can we add a `first_throttled` attribute to `Prop::RateLimited`?

Comment (grosser) [inline lib/prop/limiter.rb]: yeah that could be useful too

Comment (grosser) [inline lib/prop/limiter.rb]: added

File: lib/prop/rate_limited.rb [hunk 2]
```diff
@@ -1,23 +1,21 @@
...
 
     def initialize(options)
-      handle    = options.fetch(:handle)
-      cache_key = options.fetch(:cache_key)
-      interval  = options.fetch(:interval).to_i
-      strategy  = options.fetch(:strategy)
-
-      super(strategy.threshold_reached(options))
-
+      self.handle    = options.fetch(:handle)
```
Comment (melvinram) [inline lib/prop/rate_limited.rb]: self is implied, no?

Comment (grosser) [inline lib/prop/rate_limited.rb]: need it for assignments, otherwise it's just a local variable

**Comment by ggrossman** [comment]
:+1:

**Pull request merged**